File: Problems/Algorithms/409. Longest Palindrome/LongestPalindrome.java
class Solution {
    public int longestPalindrome(String s) {
        int[] count = new int[128];
        for (char ch : s.toCharArray()) {
            count[ch]++;
        }
        
        int ans = 0;
        for (int num : count) {
            ans += num / 2 * 2;
            if (ans % 2 == 0 && num % 2 == 1) {
                ans++;
            }
        }
        
        return ans;
    }
}

File: Problems/Algorithms/409. Longest Palindrome/longest_palindrome.rs
impl Solution {
    pub fn longest_palindrome(s: String) -> i32 {
        let mut counts = vec![0; 128];
        for &byte in s.as_bytes() {
            counts[byte as usize] += 1;
        }
        
        let mut ans = 0;
        for n in counts.iter() {
            ans += n;
            if n % 2 != 0 {
                ans -= 1;
                if ans % 2 == 0 {
                    ans += 1;
                }
            }
        }
        ans
    }
}

